src/python/pants/backend/codegen/avro/target_types.py

```diff
@@ -81,55 +74,22 @@ class AvroSourcesOverridesField(OverridesField):
     )
 
 
-class AvroSourcesGeneratorTarget(Target):
+class AvroSourcesGeneratorTarget(TargetFilesGenerator):
     alias = "avro_sources"
     core_fields = (
         *COMMON_TARGET_FIELDS,
         AvroDependenciesField,
         AvroSourcesGeneratingSourcesField,
         AvroSourcesOverridesField,
     )
-    help = "Generate a `avro_source` target for each file in the `sources` field."
-
-
-class GenerateTargetsFromAvroSources(GenerateTargetsRequest):
-    generate_from = AvroSourcesGeneratorTarget
-
-
-@rule
-async def generate_targets_from_avro_sources(
-    request: GenerateTargetsFromAvroSources,
-    files_not_found_behavior: FilesNotFoundBehavior,
-    union_membership: UnionMembership,
-) -> GeneratedTargets:
-    sources_paths = await Get(
-        SourcesPaths, SourcesPathsRequest(request.generator[AvroSourcesGeneratingSourcesField])
-    )
-
-    all_overrides = {}
-    overrides_field = request.generator[OverridesField]
-    if overrides_field.value:
-        _all_override_paths = await MultiGet(
-            Get(Paths, PathGlobs, path_globs)
-            for path_globs in overrides_field.to_path_globs(files_not_found_behavior)
-        )
-        all_overrides = overrides_field.flatten_paths(
-            dict(zip(_all_override_paths, overrides_field.value.values()))
-        )
-
-    return generate_file_level_targets(
-        AvroSourceTarget,
-        request.generator,
-        sources_paths.files,
-        union_membership,
-        # Note: Avro files cannot import from other Avro files, so do not add dependencies.
-        add_dependencies_on_all_siblings=False,
```

Comment:
I think that "dependencies on all siblings" should be rare to set, and unlike the inverse it's challenging to detect that you've set it (because dependencies are automatically more coarse-grained than they should be).

For example: `scala` has accidentally been setting this via `add_dependencies_on_all_siblings`, without us realizing (or remembering, at least): see #14382. In the interests of making dependency inference a stronger default, and removing the boilerplate required to do the right thing, I don't think that we should require it.